### PR TITLE
ci: test library installation

### PR DESCRIPTION
**Pull request by nizox**
Install and test loading the ddtrace library on every supported platform during the build process:
* Linux glibc x86_64
* Linux glibc i686
* Linux glibc arm64
* Linux musl x86_64 (Alpine Linux)
* MacOS x86_64
* Windows AMD64
* Windows x86

This PR removes the Alpine test build on CircleCI and replaces it with testing the Alpine install from the source distribution.

### CODE REVIEW

**Review by Kyle-Verhoog — APPROVED** [review]

File: .github/workflows/build_deploy.yml [hunk 1]
```diff
@@ -43,11 +51,14 @@ jobs:
...
           # configure cibuildwheel to build native archs ('auto'), and some
           # emulated ones
-          CIBW_ARCHS_LINUX: auto aarch64
-          CIBW_ARCHS_MACOS: x86_64 universal2
+          CIBW_ARCHS: ${{ matrix.archs }}
           # PyPY is not supported
           # Windows 2.7 wheels no longer can be built https://github.com/DataDog/dd-trace-py/pull/2344
           CIBW_SKIP: pp* cp27-win*
+          # Run a smoke test on every supported platform
+          CIBW_TEST_COMMAND: python {project}/tests/smoke_test.py
```
Comment (brettlangdon) [inline .github/workflows/build_deploy.yml:59]: cool!

File: tests/smoke_test.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import sys
+
+import ddtrace.bootstrap.sitecustomize as module
```
Comment (brettlangdon) [inline tests/smoke_test.py:3]: this is probably enough for initial version, but I wonder what else we can/should do for testing?

like, maybe profiling Cython code compiles, but we should test loading it too?

**Review by brettlangdon — DISMISSED** [review]

File: tests/smoke_test.py (hunk 2 above)
Comment (nizox) [reply tests/smoke_test.py:3]: Definitely! Is there a module we can import to test cython code is correctly loading?

Comment (brettlangdon) [reply tests/smoke_test.py:3]: This is a good question.... I am ok to defer for now, at least we have a way to do it.

Loading and using the msgpack encoder, generating a random id, or start the profiler could all be helpful ?

**Review by brettlangdon — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]
terrific!

**Comment by brettlangdon** [comment]
@Mergifyio backport 0.51
@Mergifyio backport 0.50

File: .github/workflows/build_deploy.yml (hunk 1 above)
Comment (Kyle-Verhoog) [reply .github/workflows/build_deploy.yml:59]: oh we should totally `DD_TESTING_RAISE` here! (And maybe do a tracer.trace()/profiler.start() as well)

**Pull request merged**